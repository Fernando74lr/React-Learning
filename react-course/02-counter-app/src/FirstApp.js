// import { Fragment } from "react";
import PropTypes from 'prop-types';

const FirstApp = ({ greeting, age, subtitle }) => {

    return (
        <>
            <h1>{ greeting }</h1>
            <h2>I'm { age } years old</h2>
            <h3>{ subtitle }</h3>
            {/* <pre>{ JSON.stringify(greeting, null, 3) }</pre> */}
            <p>My first application</p>
        </>
    );
}

// Definition of the type properties.
FirstApp.propTypes = {
    greeting: PropTypes.string.isRequired,
    age: PropTypes.number
}

// Definition of the default properties.
FirstApp.defaultProps = {
    subtitle: 'I\'m a subtitle'
}

export default FirstApp;