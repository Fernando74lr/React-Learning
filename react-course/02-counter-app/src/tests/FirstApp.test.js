// import { render } from "@testing-library/react";
// import React from 'react';
import FirstApp from "../FirstApp";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

describe('Tests in <FirstApp />', () => {
    // test('should show the message: "I\'m Fernando"', () => {
    //     const greeting = 'I\'m Fernando';

    //     const { getByText} = render(<FirstApp greeting={greeting} />);

    //     expect(getByText(greeting)).toBeInTheDocument();
    // });

    test('should show <FirstApp /> correctly', () => {
        const greeting = 'I\'m Fernando';
        const wrapper = shallow(<FirstApp greeting={greeting} />);
        expect(wrapper).toMatchSnapshot();
    });    
    
});
