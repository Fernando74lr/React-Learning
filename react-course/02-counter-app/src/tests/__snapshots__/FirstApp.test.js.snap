// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    I'm Fernando
  </h1>
  <h2>
    I'm 
    20
     years old
  </h2>
  <h3>
    I'm a subtitle
  </h3>
  <p>
    My first application
  </p>
</Fragment>
`;
